esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -476,20 +502,102 @@ bool CWsESDLConfigEx::onPublishESDLDefinition(IEspContext &context, IEspPublishE
         unsigned newseq = 0;
         StringBuffer msg;
 
-        Owned<IPropertyTree> queryRegistry = getESDLDefinitionRegistry(service.get(), false);
-
-        if (queryRegistry != NULL)
-        {
-            addESDLDefinition(queryRegistry, service.get(), serviceXMLTree.get(), newqueryid, newseq, user, deletePrevious);
-            if (newseq)
-                resp.setEsdlVersion(newseq);
+        {   // We don't need the queryregistry around after the addESDLDefinition.
+            Owned<IPropertyTree> queryRegistry = getESDLDefinitionRegistry(service.get(), false);
 
-            msg.appendf("Successfully published %s", newqueryid.str());
+            if (queryRegistry != NULL)
+            {
+                addESDLDefinition(queryRegistry, service.get(), serviceXMLTree.get(), newqueryid, newseq, user, deletePrevious);
+                if (newseq)
+                    resp.setEsdlVersion(newseq);
+            }
+            else
+            {
+                msg.set("Could not publish ESDL Definition, unable to fetch ESDL Definition registry.");
+                resp.updateStatus().setCode(-1);
+            }
         }
-        else
+
+        msg.appendf("Successfully published %s", newqueryid.str());
```

Comment:
In line 516, msg says "Could not publish...". But, "Successfully published..." appended here?